mlflow/utils/file_utils.py

```diff
@@ -384,6 +389,8 @@ def local_file_uri_to_path(uri):
     No-op if the uri does not have the expected scheme.
     """
     path = urllib.parse.urlparse(uri).path if uri.startswith("file:") else uri
+    if os.name == 'nt' and not path:
```

Comment:
To reduce my own ignorance :P do you have an example (maybe one of our tests fails without this?) that illustrates the need for this change? What sort of input do we pass to this function that requires special handling for Windows / why does the solution in https://stackoverflow.com/a/43925228/12111347 (which seems to be what we had before) not work?

Reply:
![image](https://user-images.githubusercontent.com/51172624/91600916-9ae14780-e91d-11ea-9a44-64af3c57307b.png)

Here is one example:

I believe `urlparse` incorrectly parses the windows path as an entire `netloc` instead of a `path`.

Reply:
Windows `\`-style paths aren't valid URIs. Afaik, the guidance for windows users has been to use file:/// URIs instead of backslashed paths with MLflow. I may be mistaken, though